services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -26,17 +25,17 @@ public class AwsConnectionServiceImpl implements AwsConnectionService {
     private AwsConnectionRepo awsConnectionRepo;
 
     @Override
-    public AwsDatasourceResponse saveConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, Projects project) throws InvalidConnectionTypeException {
+    public AwsDatasourceResponse saveConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, String userName) throws InvalidConnectionTypeException {
 
-        Optional<AWSDatasource> dataSources = awsConnectionRepo.findByDataSourceName(awsDatasourceRequest.getDataSourceName());
+        Optional<AWSDatasource> dataSources = awsConnectionRepo.findByDataSourceNameIgnoreCaseAndActive(awsDatasourceRequest.getDataSourceName().trim(),true);
```

Comment:
this logic should exclude the passed id and check the name exists in other records in table

Reply:
used findByDataSourceNameIgnoreCase().